File: staGledas.Model/Models/KlubKomentari.cs
using System;

namespace staGledas.Model.Models
{
    public class KlubKomentari
    {
        public int Id { get; set; }
        public int ObjavaId { get; set; }
        public int KorisnikId { get; set; }
        public string? Sadrzaj { get; set; }
        public DateTime DatumKreiranja { get; set; }
        public int? ParentKomentarId { get; set; }
        public virtual KlubObjave? Objava { get; set; }
        public virtual Korisnici? Korisnik { get; set; }
        public virtual KlubKomentari? ParentKomentar { get; set; }
    }
}

File: staGledas-UI/stagledas_mobile/lib/models/klub_komentari.dart
import 'package:json_annotation/json_annotation.dart';
import 'korisnik.dart';

part 'klub_komentari.g.dart';

@JsonSerializable()
class KlubKomentari {
  int? id;
  int? objavaId;
  int? korisnikId;
  String? sadrzaj;
  DateTime? datumKreiranja;
  int? parentKomentarId;
  Korisnik? korisnik;

  KlubKomentari({
    this.id,
    this.objavaId,
    this.korisnikId,
    this.sadrzaj,
    this.datumKreiranja,
    this.parentKomentarId,
    this.korisnik,
  });

  String get displayName => korisnik?.korisnickoIme ?? 'Korisnik';
  bool get isReply => parentKomentarId != null;

  factory KlubKomentari.fromJson(Map<String, dynamic> json) =>
      _$KlubKomentariFromJson(json);

  Map<String, dynamic> toJson() => _$KlubKomentariToJson(this);
}
